structure/linkedlist/doubly.go

```diff
@@ -64,14 +64,14 @@ func (ll *Doubly) DelAtBeg() any {
 	if ll.Head != nil {
 		ll.Head.Prev = nil
 	}
-	return cur.Val
+	return cur.Val, true
 }
 
 // DetAtEnd Delete a node at the end of the linkedlist
-func (ll *Doubly) DelAtEnd() any {
+func (ll *Doubly[T]) DelAtEnd() (T, bool) {
 	// no item
 	if ll.Head == nil {
-		return -1
+		return *new(T), false
```

Comment:
This is actually quite confusing; I would prefer something like this instead
```suggestion
		var r T
		return r, false
```

For the first couple of minutes I was confused that I didn't see any `new` function then I remembered that it was the inbuilt new.

Reply:
Sure, I used that syntax for brevity but it's most likely faster using the `var r T` syntax since it'll keep the value on the stack